avro-serializer/src/main/java/io/confluent/kafka/serializers/KafkaAvroDeserializer.java

```diff
@@ -53,6 +54,11 @@ public Object deserialize(String s, byte[] bytes) {
     return deserialize(bytes);
   }
 
+  /**
+   * Pass a reader schema to get an Avro projection
+   */
+  public Object deserialize(String s, byte[] bytes, Schema readerSchema) { return deserialize(bytes, readerSchema); }
```

Comment:
Yes, that sounds right to me.

I think by "dynamic lookup", you mean [this code](https://github.com/confluentinc/schema-registry/blob/master/avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java#L202-L223), which does makes sense to keep for cases where you don't want a projection.

As you said, this has no impact on regular Kafka users of the deserializers and sets the stage for latter being able to specify a different reader schema.